### DISCUSSION TITLE
How to get dual value if used Minimum Cost Flows?

### DISCUSSION THREAD

**Discussion by Robot-1x** [google/or-tools#3322] [Graph and Algorithms]
What version of OR-Tools and what language are you using?
Version: 9.2.9972
Language: Java

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)
ortools(MinCostFlow)

What operating system (Linux, Windows, ...) and version?
MacOS 11.6.5

What did you do?
To solve a pure network model and use the dual value. As the network model scale is too large（too many vertices/edges）, open source linear programming solver can not work or works too long. I used MinCostFlow(push-relabel). It works and output optimal solution. Hope to know how to calculate the dual value of the primal problem.
![image](https://user-images.githubusercontent.com/25146455/171398785-b3aae921-03bc-4c74-990a-6924d2909e52.png)

**Comment by lperron** [comment]
The API does not supports it.
I am not even sure this is feasible from the algorithm

**Comment by Robot-1x** [comment]
I see some explanation of min_cost_flow.h. I'm not sure this is same with what I know as used simplex method and get dual value.
![image](https://user-images.githubusercontent.com/25146455/171415612-2300586a-0ba9-451e-8366-52d116f8eaeb.png)
[https://google.github.io/or-tools/cpp_graph/min__cost__flow_8h_source.html](url)
